### DISCUSSION TITLE
Image - imports and exports

### DISCUSSION THREAD

**Discussion by BoBoBaSs84** [MadDeCoDeR/BFG-Resource-File-Manager#10] [Ideas]
**What formats should be supported for export?**
Currently `tga` is the standard export format.
Maybe `png` and `jpg`?

**What formats should be supported for import?**
Currently no format is possible to import.
Maybe `dds`?

**Comment by MadDeCoDeR** [comment]
Both Import and Export must support the same format. The reason for that is that you might want to export a binarized file edit it and then import and re-binirize it

**Reply by BoBoBaSs84** [reply]
So when exporting to `tga` we either loose the mipmaps or export them all? (Currently one [the main] gets exported.)

```csharp
...
//TODO Multilevel Textures
img = CovertToBitmap(images[0], texType, format, colorFormat);
...
```

And when importing we use the "one" main `tga` and create the mipmaps?
Is there a mipmap limit?

Sorry for all the questions (-.-)

**Reply by MadDeCoDeR** [reply]
The engine is usually generating the mipmaps when binarizing the image.
Each mipmap is usually smaller than the previous one. I belive there is a limit to how small an image can be, also the resolution is divided by the previous one

**Comment by MadDeCoDeR** [comment]
The reason why I chose tga is because the engine expects tga images to load and binirize them
